frontend/src/contexts/RecordContextProvider.js

```diff
@@ -28,8 +28,19 @@ export default function RecordContextProvider({children}) {
             .then((newRecord) => setRecords([...records, newRecord]))
             .catch(console.log);
 
+    const deleteRecord = (id) =>
+        removeRecord(id, token)
+            .then(() => setRecords(records.filter((record) => record.id !== id)))
+            .catch(console.log);
+
+    const editRecord = (id, titel, recordLink, tagsList, description, publicStaus) =>{
+        updateRecord(id, titel, recordLink, tagsList, description, publicStaus, token)
+            .then((updatedRecord) => setRecords([...records.filter((record) => record.id !== updatedRecord.id), updatedRecord]))
+            .catch(console.log)
+    }
+
```

Comment:
I think you only need the `.catch` statements if you have no security token. Then you would catch the error in the `console.log`. Now it is the security context that catches the error.